wideq/core.py

```diff
@@ -240,166 +345,218 @@ def parse_oauth_callback(url):
-                   data.get('language', DEFAULT_LANGUAGE))
-
 
 class Auth(object):
-    def __init__(self, gateway, access_token, refresh_token):
+    def __init__(self, gateway, oauth_url, access_token, refresh_token, user_number):
         self.gateway = gateway
+        self.oauth_url = oauth_url
         self.access_token = access_token
         self.refresh_token = refresh_token
+        self.user_number = user_number
 
     @classmethod
     def from_url(cls, gateway, url):
         """Create an authentication using an OAuth callback URL.
         """
+        oauth_url, auth_code, user_number = parse_oauth_callback(url)
+        access_token, refresh_token = login(oauth_url, auth_code)
 
-        access_token, refresh_token = parse_oauth_callback(url)
-        return cls(gateway, access_token, refresh_token)
+        return cls(gateway, oauth_url, access_token, refresh_token, user_number)
 
-    def start_session(self) -> Tuple['Session', List[Dict[str, Any]]]:
+    def start_session(self):
         """Start an API session for the logged-in user. Return the
         Session object and a list of the user's devices.
         """
-
-        session_info = login(self.gateway.api_root, self.access_token,
-                             self.gateway.country, self.gateway.language)
-        session_id = session_info['jsessionId']
-        return Session(self, session_id), get_list(session_info, 'item')
+        return Session(self)
 
     def refresh(self):
         """Refresh the authentication, returning a new Auth object.
         """
 
-        new_access_token = refresh_auth(self.gateway.oauth_root,
-                                        self.refresh_token)
-        return Auth(self.gateway, new_access_token, self.refresh_token)
+        new_access_token = refresh_auth(self.oauth_url, self.refresh_token)
+        return Auth(
+            self.gateway,
+            self.oauth_url,
+            new_access_token,
+            self.refresh_token,
+            self.user_number,
+        )
 
-    def serialize(self) -> Dict[str, str]:
+    def dump(self):
         return {
-            'access_token': self.access_token,
-            'refresh_token': self.refresh_token,
+            "access_token": self.access_token,
+            "refresh_token": self.refresh_token,
+            "oauth_url": self.oauth_url,
+            "user_number": self.user_number,
         }
 
+    @classmethod
+    def load(cls, gateway, data):
+        return cls(
+            gateway,
+            data["oauth_url"],
+            data["access_token"],
+            data["refresh_token"],
+            data["user_number"],
+        )
+
 
 class Session(object):
-    def __init__(self, auth, session_id) -> None:
+    def __init__(self, auth, session_id=None):
         self.auth = auth
         self.session_id = session_id
 
     def post(self, path, data=None):
-        """Make a POST request to the API server.
+        """Make a POST request to the APIv1 server.
```

Comment:
Is this really for thinq1? If so, let's delete it.

Reply:
As I said before, this is used, depend on the device